src/main/java/eu/adlogix/appnexus/oas/client/service/impl/DefaultCampaignService.java

```diff
@@ -0,0 +1,139 @@
+package eu.adlogix.appnexus.oas.client.service.impl;
+
+import static eu.adlogix.appnexus.oas.client.utils.ValidatorUtils.checkNotEmpty;
+import static eu.adlogix.appnexus.oas.client.utils.ValidatorUtils.checkNotNull;
+import static org.apache.commons.lang3.StringUtils.defaultIfEmpty;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import eu.adlogix.appnexus.oas.client.domain.Campaign;
+import eu.adlogix.appnexus.oas.client.domain.StatefulDomainManager;
+import eu.adlogix.appnexus.oas.client.domain.enums.BillTo;
+import eu.adlogix.appnexus.oas.client.domain.enums.CampaignType;
+import eu.adlogix.appnexus.oas.client.parser.XmlToCampaignParser;
+import eu.adlogix.appnexus.oas.client.service.AbstractOasService;
+import eu.adlogix.appnexus.oas.client.service.CampaignService;
+import eu.adlogix.appnexus.oas.client.service.OasApiService;
+import eu.adlogix.appnexus.oas.client.transform.CampaignCreateParameterMapTransformer;
+import eu.adlogix.appnexus.oas.client.transform.CampaignParameterMapTransformer;
+import eu.adlogix.appnexus.oas.client.transform.CampaignUpdateParameterMapTransformer;
+import eu.adlogix.appnexus.oas.client.xml.ResponseParser;
+import eu.adlogix.appnexus.oas.client.xml.XmlRequestGenerator;
+
+/**
+ * Default implementation of {@link CampaignService} which provides functions
+ * for Campaign related operations
+ * 
+ */
+public class DefaultCampaignService extends AbstractOasService implements CampaignService {
+
+	private final XmlRequestGenerator readCampaignRequestGenerator = new XmlRequestGenerator("read-campaign");
+	private final XmlRequestGenerator addCampaignRequestGenerator = new XmlRequestGenerator("add-campaign");
+	private final XmlRequestGenerator updateCampaignRequestGenerator = new XmlRequestGenerator("update-campaign");
+
+	public DefaultCampaignService(OasApiService apiService) {
+		super(apiService);
+	}
+
+	/**
+	 * Retrieves an existing {@link Campaign} by the given campaign id
+	 * 
```

Comment:
we should just set the annotation @override ... recopying the doc is overkill and error prone (in case the doc changes). It has also the advantage of the compiler checking to make sure you actually are overriding a method when you think you are. This way, if you make a common mistake of misspelling a method name or not correctly matching the parameters, you will be warned that you method does not actually override as you think it does. Secondly, it makes your code easier to understand because it is more obvious when methods are overwritten.